core/src/main/scala/stryker4s/run/ProcessMutantRunner.scala

```diff
@@ -48,14 +48,16 @@ class ProcessMutantRunner(command: Command, process: ProcessRunner)(implicit con
 
     val totalMutants = mutatedFiles.flatMap(_.mutants).size
 
-    val runResults = for {
+    val mutantsWithPath = for {
       mutatedFile <- mutatedFiles
       subPath = mutatedFile.fileOrigin.relativePath
       mutant <- mutatedFile.mutants
-      id = mutatedFiles.flatMap(_.mutants).toSeq.indexOf(mutant) + 1
-    } yield {
+    } yield (mutant, subPath)
+
+    val runResults = mutantsWithPath.zipWithIndex.map{ case ((mutant, subPath), index) =>
```

Comment:
I couldn't get it to work within the for-comprehension, so this was the cleanest rewrite I could think of.

Reply:
You should be able to do this:
```scala
val mutantsWithPath = for {
      (mutatedFile, index) <- mutatedFiles.zipWithIndex
      subPath = mutatedFile.fileOrigin.relativePath
      mutant <- mutatedFile.mutants
    } yield {
```

Reply:
Wouldn't that give an index per file instead of per mutation?

Reply:
But this generates an index for each file, but we need an index for each mutant. With this code all mutants from the same file get the same id.

Reply:
Oow right didn't think of that.

Reply:
A better fix might be to assign IDs before excluding mutations

Reply:
Right now the mutants are first created in MutantMatcher.scala and then excluded in MutantFinder.scala. So excluding logic would have to be built in MutantMatcher.scala. But I'm not sure if that's clean. 

Implicitly passing the Config to MutantMatchers works for example. Then we can do the exclusion logic there. Example:
```
def ~~>(mutated: MethodMutator, f: String => Term)(implicit config: Config): Seq[Mutant] = {
  if (config.excludedMutations.exclusions.contains(mutated.mutationName))
    Mutant(-1, original, mutated.apply(f), mutated) :: Nil // Excluded
  else
    Mutant(stream.next, original, mutated.apply(f), mutated) :: Nil
}
```

This gives the included mutators correct indexes: 0, 1, 2, 3, etc. and all excluded -1. (obviously -1 is a bad idea but I was just trying it out)

This feels like logic that shouldn't be in this class. But I could be wrong. What do you think @hugo-vrijswijk ?

Reply:
I'm not sure. The MutantMatcher taking user configuration into consideration for what mutants to match on makes sense to me